### PR TITLE
blog post for knative release 0.22

### LINKED ISSUE #3485: Blog post for Knative release v0.22
Blog post for Knative release v0.22

Add post to https://knative.dev/blog/releases/

### PR DESCRIPTION
**Pull request by csantanapr**
blog post for knative release 0.22

Closes #3485

### CODE REVIEW

File: blog/releases/announcing-knative-v0-22-release.md
Comment (RichardJJG) [inline blog/releases/announcing-knative-v0-22-release.md:16]:
```suggestion
To upgrade, follow the steps in [Upgrading Your Installation](https://knative.dev/docs/install/upgrade-installation/) for the relevant components.
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (pmbanugo) [inline blog/releases/announcing-knative-v0-22-release.md:32]:
```suggestion
- Serving Domain Mapping improves multi-tenancy support to avoid domain name 'squatting' in the cluster.
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (pmbanugo) [inline blog/releases/announcing-knative-v0-22-release.md:34]: I don't understand this sentence. Maybe because of my lack of understanding of Eventing.

Comment (pmbanugo) [inline blog/releases/announcing-knative-v0-22-release.md:34]:
```suggestion
- Eventing Kafka broker now Kubernetes's minimum version is 1.18
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (pmbanugo) [inline blog/releases/announcing-knative-v0-22-release.md:35]:
```suggestion
- Eventing Kafka now supports the ability to choose between ordered and unordered delivery
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (pmbanugo) [inline blog/releases/announcing-knative-v0-22-release.md:37]:
```suggestion
- There are two new CLI plugins to align with the 0.22 release, [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) and [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (pmbanugo) [inline blog/releases/announcing-knative-v0-22-release.md:38]:
```suggestion
- The Knative Operator was released with some bug fixes and supports the latest version of Knative Serving and Eventing.
```

**Comment by csantanapr** [comment]
/hold
Wow! is excellent to see PR reviews already over might. Thank you @RichardJJG @pmbanugo 👍 

I will make the changes. I prefer to click the button to accept suggested changes, but then it is a pain to deal with the Google CLA with multiple authors.

**Comment by abrennan89** [comment]
@csantanapr when it's all ready Grant should be able to manually fix the CLA if it's a problem, probably faster to do that than for you to have to manually do the suggestions 🙂

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:3]: Do we need "Version" and "v" in the title?

**Comment by csantanapr** [comment]
@abrennan89 Im already aware that reaching anyone from the Google company can fix the CLA problem. 

But I choose to not go thru that route and that this problem is a that I consider very bad taste for open source. I don't want to project or give the impression that I'm fine with it.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:32]: Don't use 'squatting' here; explain properly what it means

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:33]:
```suggestion
- Eventing now allows subscribers and triggers from different namespaces to be used together.
```

Not totally sure if this is right or what this means otherwise, it seems like you're saying triggers can have subscribers that are in a different namespace to them now?

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:34]:
```suggestion
- 1.18 is now the minimum Kubernetes version required to use the Apache Kafka broker with Knative Eventing v0.22.
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [reply blog/releases/announcing-knative-v0-22-release.md:35]: Use Apache Kafka everywhere, or just "Kafka with Knative Eventing"?

Comment (abrennan89) [reply blog/releases/announcing-knative-v0-22-release.md:35]: Also, be consistent about using periods / full stops at the end of sentences 😉

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:37]:
```suggestion
- Version 0.22 of the `kn` CLI contains bug fixes and minor feature enhancements.
- A breaking change was required to align the client API with the Kubernetes client API.
- There are two new CLI plugins available in version 0.22; [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) and [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka).
```

Comment (abrennan89) [reply blog/releases/announcing-knative-v0-22-release.md:37]: @omerbensaadon @snneji @RichardJJG can we choose one way to consistently refer to "version" across docs and Knative in general? Do you have any preference? I don't want to keep using "version 0.22" in some places and "v0.22" in others for example if we can help it. Can you maybe take this to the TOC, Omer?

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:38]:
```suggestion
- The Knative Operator 0.22 release contains bug fixes, and supports version 0.22 of Knative Serving and Knative Eventing.
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:46]:
```suggestion
- Added the `autocreateClusterDomainClaims` flag to the network ConfigMap. [networking/#330](https://github.com/knative/networking/pull/330)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:50]:
```suggestion
- Added validation that a default `max-scale` is set if a `max-scale-limit` is specified in the autoscaler ConfigMap. Otherwise, the default `max-scale`, `0` = no max, would fail validation as it is above the `max-scale-limit`. [#10921](https://github.com/knative/serving/pull/10921)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:51]:
```suggestion
- Raised the resource request and limits of the autoscaler to 100m/100Mi, 1000m/1000Mi respectively. [#10865](https://github.com/knative/serving/pull/10865)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:52]:
```suggestion
- Fixed a regression where the latency of starting a pod could be 10s or more. [#10992](https://github.com/knative/serving/pull/10992)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:53]:
```suggestion
- Reduced required memory allocations in the activator significantly, particularly when tracing is turned off. [#11016](https://github.com/knative/serving/pull/11016), [#11013](https://github.com/knative/serving/pull/11013), [#11009](https://github.com/knative/serving/pull/11009), [#11008](https://github.com/knative/serving/pull/11008)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:54]:
```suggestion
- Fixed the incorrect gateway name format for the domain mapping auto TLS feature for `net-istio` implementation. [net-istio#532](https://github.com/knative-sandbox/net-istio/pull/532)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:61]:
```suggestion
- `v1alpha1` channel duck types are no longer supported. [#5005](https://github.com/knative/eventing/pull/5005)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:62]: Again, it's not clear to me what you're trying to say here re "set another namespace's Subscriber".

From a docs perspective;
- Please use lowercase for triggers, subscribers
- Do no use possessive ('s) for namespace and other inanimate objects. E.g. "If the namespace for a subscriber has not been set, the subscriber defaults to using the namespace of the trigger."

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:70]:
```suggestion
- Fixed a bug that caused the PingSource adapter to not always release the leader election lease when terminating. [#5162](https://github.com/knative/eventing/pull/5162)
- Fixed a bug that caused PingSource objects to send duplicate events when the leading adapter failed to renew its lease. [#4908](https://github.com/knative/eventing/pull/4908)
- Fixed a  bug that prevented namespace-scoped InMemoryChannel objects from becoming `READY`. [#4906](https://github.com/knative/eventing/pull/)
- Fixed excessive number of replicas that were being created for `eventing-webhook`. [#5112](https://github.com/knative/eventing/pull/5112)
- The webhook timeout for API server calls has been increased to 10 seconds. [#5175](https://github.com/knative/eventing/pull/5175)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [reply blog/releases/announcing-knative-v0-22-release.md:74]: Be consistent with PingSource formatting.

@omerbensaadon this is part of the API object formatting issue for the style guide; we need to make a decision on that ASAP so that folks have some guidance.

Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:74]:
```suggestion
- Do not set a finalizer for PingSource controllers. Use `reconcilekind` instead. [#5002](https://github.com/knative/eventing/pull/5002)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:79]: Decide on consistent name here, @matzew any thoughts? Should this be Apache Kafka broker instead?

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:83]: I'd move this to way earlier in the doc; also we call this out separately earlier in the doc for the Kafka broker, so I'm not sure we need to have it in two places? The headings seem a bit messy in this blog in general, there isn't a very logical flow IMO.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:89]: Again, I feel like this was mentioned earlier but in less detail. We need a better overall structure for this doc, but we can follow that up in a future release. Won't block the PR over it.

Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:89]:
```suggestion
- The Kafka broker now supports ordered delivery. You can choose between ordered and unordered delivery by using the `kafka.eventing.knative.dev/delivery.order` label in the trigger spec. See the [Kafka broker](https://knative.dev/docs/eventing/broker/kafka-broker) documentation. [#589](https://github.com/knative-sandbox/eventing-kafka-broker/pull/)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:94]:
```suggestion
- Added a producer interceptor, `io.cloudevents.kafka.PartitionKeyExtensionInterceptor`, to provide ordered delivery based on the partitioning extension of the CloudEvents spec. [#751](https://github.com/knative-sandbox/eventing-kafka-broker/pull/751)
- Fix unable to deploy KafkaSink without Kafka Broker installed [#714](https://github.com/knative-sandbox/eventing-kafka-broker/pull/714)
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:98]: This is mentioned already higher up in the doc, delete entirely.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:101]: Move this to earlier in the doc, also the META heading is not clear. I don't think this needs a whole heading, could probably just go in the highlights section.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:106]:
```suggestion
This release adds support for CRUD management of domain mappings. You can use the `kn domain` CLI command, along with its sub-commands `create`, `update`, `describe`, `list`, and `delete`, to fully manage DomainMapping resources to use custom domains for Knative services.
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:121]: @csantanapr can you please open a docs issue for `kn domain` to get it properly documented in the site so it's not only explained here?

@rhuss can ya'll in the CLI group please come up with a process to make sure changes to the CLI like this are being communicated properly to docs? Otherwise they just don't get documented. CC @omerbensaadon

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:125]: Be consistent with "Client API", it's "client API" earlier in the doc, please choose one and use throughout.

Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:125]:
```suggestion
A `context.Context` object was added as the first argument to every Client API method to align with the Kubernetes client API signatures. This change does not affect client CLI usage unless the client-specific Golang API used to communicate with the Knative backend has changed.

To migrate to the updated API signature, you can pass an already existing context to the call, or use one of the available standard contexts, such as `context.TODO()`.
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:136]: The new plugins are mentioned more generally in a section earlier in the doc; let's group this information maybe? It seems confusing to have it in different places.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:142]:
```suggestion
### Minor CLI updates

- `kn export` now uses the _Export_ format as the default.
- Added `S` column to specify built-in sources in `kn source list-types`.
- Added support for namespaces for all commands that takes a `--sink` option.
```

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [inline blog/releases/announcing-knative-v0-22-release.md:151]:
```suggestion
- Delete installed ingress resources only. [#548](https://github.com/knative/operator/pull/548)
- Allow the update of ingress resources with `spec.additionalManifests`. [#531](https://github.com/knative/operator/pull/531)
- Refactored the cache mechanism. [#532](https://github.com/knative/operator/pull/532)
- Filter redundant resources in the target manifest. [#509](https://github.com/knative/operator/pull/509)
```

**Review by abrennan89 — CHANGES_REQUESTED** [review]
@csantanapr can we make a proper template for this going forward, my biggest issue with it is that the information seems very randomly organized, e.g. many different CLI sections throughout the doc with repeated info.
Won't block the PR on this but have added some suggestions.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (omerbensaadon) [reply blog/releases/announcing-knative-v0-22-release.md:74]: If you make the decision on this, I will document it :D

Comment (omerbensaadon) [reply blog/releases/announcing-knative-v0-22-release.md:74]: CC: @RichardJJG @snneji who are also qualified to make a decision on this

File: blog/releases/announcing-knative-v0-22-release.md
Comment (rhuss) [reply blog/releases/announcing-knative-v0-22-release.md:129]: Sorry for that. I will update the PR template to raise awareness, so that doc relevant PR should be properly communicated.

What would be the best way to communicate ? A ping within the PR or creating an issue in the docs repo ?

Comment (abrennan89) [reply blog/releases/announcing-knative-v0-22-release.md:129]: @rhuss creating an issue in the docs repo would be perfect, thank you so much 🙂

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:144]: They are mentioned in the first section `Highlights` in one sentence.
```
- There are two new CLI plugins align with 0.22 release, [kn-plugin-admin](https://github.com/knative-sandbox/kn-plugin-admin) and [kn-plugin-source-kafka](https://github.com/knative-sandbox/kn-plugin-source-kafka)
```

This section is the area with the content details about the plugins. 

Let me know if you want me to remove the sentence from the first section `Highlights`

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:133]: Just to make you aware I didn't write this content. 
Is a copy and paste verbatim from the release notes, this blog post is just trying to put all existing release notes in a single place. The only change I do is some minor adjustments. 

https://github.com/knative/client/releases/tag/v0.22.0
image

I think is @rhuss who writes the release notes for client

Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:133]: I'm glad that tech writers are reviewing this notes, I will be happy to make any suggested changes, and then I can open an issue to the respective repos for someone with write access to release notes to update them there.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:129]: issues https://github.com/knative/docs/issues/3496 opened @abrennan89

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:109]: I agree I removed it from the doc, is more internal info

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md]: remove left it in the highlights above

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:89]: I remove it from here and leave it in the earlier "highlights" section

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:83]: I will go calling the broker "Apache Kafka Broker" and follow consistency with what we already have on the website https://knative.dev/docs/eventing/broker/kafka-broker/

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:66]: I really don't know what is this trying to say either is not clear to me.
I copy from the release notes here 
https://github.com/knative/eventing/releases/tag/v0.22.1

image

I think what it means is that now Triggers allow specifying a Subscriber in a different namespace, when the 
namespace of the Subscriber is not specified it will default to the same namespace as the Trigger.

@n3wscott is my description accurately above ^^ ?

I will comment it out for now, but I didn't see the docs updated to cover this new enhancement

Comment (devguyio) [reply blog/releases/announcing-knative-v0-22-release.md:66]: Yes that's correct. Now in the trigger subscriber you can specify the namespace of the subscriber, if it's not specified then it'll default to the same namespace of the trigger.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:37]: My opinion is that deciding to use 0.22 vs. v0.22 in the website/docs, is something that the docs/ux team can make the decision and document that this is the style across the website.

I added the question in the new doc that @omerbensaadon started https://docs.google.com/document/d/1tigolRzKhronvgoAJfszE-XHWWd0LQ8o_j0D73O9KvM/edit

Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:37]: I'm making everything consistent for now to use `v0.22`

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:33]: yes, I think so

File: blog/releases/announcing-knative-v0-22-release.md
Comment (csantanapr) [reply blog/releases/announcing-knative-v0-22-release.md:3]: I will remove the small `v`

**Comment by csantanapr** [comment]
@abrennan89 I made the requested changes, it should be ready now.

File: blog/releases/announcing-knative-v0-22-release.md
Comment (abrennan89) [reply blog/releases/announcing-knative-v0-22-release.md:68]: @devguyio thanks for clarifying! If there's no docs issue to add this change could you open one please?

**Pull request merged**
